File: app/src/androidTestKiwix/java/org/kiwix/kiwixmobile/utils/StandardActions.java
/*
 * Kiwix Android
 * Copyright (C) 2018  Kiwix <android.kiwix.org>
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program. If not, see <http://www.gnu.org/licenses/>.
 */
package org.kiwix.kiwixmobile.utils;

import android.util.Log;
import com.schibsted.spain.barista.interaction.BaristaMenuClickInteractions;
import com.schibsted.spain.barista.interaction.BaristaSleepInteractions;
import org.kiwix.kiwixmobile.R;

import static androidx.test.espresso.Espresso.onData;
import static androidx.test.espresso.action.ViewActions.longClick;
import static androidx.test.espresso.matcher.ViewMatchers.withId;
import static com.schibsted.spain.barista.interaction.BaristaDialogInteractions.clickDialogPositiveButton;
import static org.kiwix.kiwixmobile.testutils.TestUtils.TEST_PAUSE_MS;
import static org.kiwix.kiwixmobile.testutils.TestUtils.getResourceString;
import static org.kiwix.kiwixmobile.testutils.TestUtils.withContent;

/**
 * Created by mhutti1 on 27/04/17.
 */

public class StandardActions {

  public static void enterSettings() {
    BaristaSleepInteractions.sleep(TEST_PAUSE_MS);
    BaristaMenuClickInteractions.clickMenu(getResourceString(R.string.menu_settings));
  }

  public static void deleteZimIfExists(String zimName, Integer adapterId) {
    try {
      onData(withContent(zimName)).inAdapterView(withId(adapterId)).perform(longClick());
      clickDialogPositiveButton();
      Log.i("TEST_DELETE_ZIM", "Successfully deleted ZIM file [" + zimName + "]");
    } catch (RuntimeException e) {
      Log.i("TEST_DELETE_ZIM", "Failed to delete ZIM file [" + zimName + "]... " +
          "Probably because it doesn't exist");
    }
  }
}

File: app/src/androidTest/java/org/kiwix/kiwixmobile/utils/StandardActions.kt
/*
 * Kiwix Android
 * Copyright (c) 2019 Kiwix <android.kiwix.org>
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program. If not, see <http://www.gnu.org/licenses/>.
 *
 */
package org.kiwix.kiwixmobile.utils

import androidx.compose.ui.test.junit4.ComposeContentTestRule
import androidx.compose.ui.test.onNodeWithTag
import androidx.compose.ui.test.performClick
import org.kiwix.kiwixmobile.core.main.CoreMainActivity
import org.kiwix.kiwixmobile.core.main.LEFT_DRAWER_SETTINGS_ITEM_TESTING_TAG
import org.kiwix.kiwixmobile.testutils.TestUtils.waitUntilTimeout

/**
 * Created by mhutti1 on 27/04/17.
 */
object StandardActions {
  fun enterSettings(composeContentTest: ComposeContentTestRule) {
    composeContentTest.apply {
      waitUntilTimeout()
      onNodeWithTag(LEFT_DRAWER_SETTINGS_ITEM_TESTING_TAG).performClick()
    }
  }

  fun openDrawer(coreMainActivity: CoreMainActivity) {
    coreMainActivity.openNavigationDrawer()
  }

  fun closeDrawer(coreMainActivity: CoreMainActivity) {
    coreMainActivity.closeNavigationDrawer()
  }
}
